libbeat/reader/syslog/syslog.go

```diff
@@ -28,23 +28,58 @@ import (
 	"github.com/elastic/elastic-agent-libs/mapstr"
 )
 
+// Note: When re-generating these files, it may be necessary to remove the
+//       //line: directives from the *_gen.go files. Otherwise, code coverage
+//       may fail to build due to an error similar to:
+//
+//         cover: inconsistent NumStmt: changed from 2 to 1
+//
 //go:generate ragel -Z -G2 -o rfc3164_gen.go parser/parser_rfc3164.rl
 //go:generate ragel -Z -G2 -o rfc5424_gen.go parser/parser_rfc5424.rl
 
 var (
-	ErrPriority       = errors.New("message has invalid or missing priority")
-	ErrTimestamp      = errors.New("message has invalid or missing timestamp")
-	ErrHostname       = errors.New("message has invalid or missing hostname")
-	ErrVersion        = errors.New("message has invalid or missing version")
-	ErrAppName        = errors.New("message has invalid or missing app name")
-	ErrProcID         = errors.New("message has invalid or missing proc ID")
-	ErrMsgID          = errors.New("message has invalid or missing msg ID")
-	ErrStructuredData = errors.New("message has invalid or missing structured data")
-	ErrSDID           = errors.New("invalid structured data ID value, expected a value")
-	ErrSDIDDuplicated = errors.New("message contains duplicated structured data ID")
-	ErrSDParam        = errors.New("invalid structured data parameter, expected a value")
+	// ErrPriority indicates a priority value is outside the acceptable range.
+	ErrPriority = errors.New("priority value out of range (expected 0..191)")
+	// ErrEOF indicates the message is truncated and cannot be parsed.
+	ErrEOF = errors.New("message is truncated (unexpected EOF)")
```

Comment:
Is there a reason to not use `io.ErrUnexpectedEOF`? It's commonly used for thing like this.

Reply:
Maybe EOF is the wrong terminology here. I don't want to confuse an improperly formatted/shaped message with not receiving enough data, but then again, how do you accurately tell the difference? I'd be okay with using the one from the standard library as long as it doesn't lose its meaning here (does it really matter saying "message is truncate"?).

Reply:
This doesn't make a lot of sense either (error that is):

```
input:

<13> test-host this is the message

error: "parsing error at position 5: unexpected EOF"
```

The actual error here is that the input doesn't match the expected shape. There should be a timestamp directly after the `<13>`, but for this test, it's missing. The parser is supposed to pull out the individual fields (timestamp, hostname, etc) and validation on those values happens in the host language, after the parser has run. If, however, the parser can't pull out all the fields, it fails completely. Aside from error-ing out, the only other thing I can think of is that whatever can't be parsed out will be put into the message field and we move on. This already happens, but I believe the original, whole message gets used. It seems a bit pointless to do that, though, and I would argue using something like grok instead.

Reply:
Yeah. Depending on how much you want to go into it, a way to do this might be to have (parse error([new error type]missing field(unexpected EOF))), which would end up looking like "parsing error at position 5: missing timestamp: unexpected EOF" when rendered. The logic would be to look for the first missing field when there is an `io.ErrUnexpectedEOF` added by the parser and interpose the missing field error, or similar.

In terms of use of the `io.ErrUnexpectedEOF` sentinel, its used generally in decoders for this kind of thing. It's best to put yourself in the mindset of a plan9 developer where _everything_ is a file to see why.